doc/user/methods/quantization.md

```diff
@@ -0,0 +1,226 @@
+
+Quantization is particularly useful for deploying models on resource-constrained devices or for scenarios where model size and inference speed are critical.
+
+#### Simple usage example:
+
+```c++
+// Quantize a pre-trained neural network model to 8-bit precision:
+
+// Load a pre-trained model (replace with your model loading code)
+mlpack::FFN<> model;
+mlpack::data::Load("model.bin", "model", model);
+
+// Create a quantizer with default settings (8-bit quantization)
+mlpack::Quantization quantizer;
+
+// Apply quantization to the model
+quantizer.Apply(model);
+
+// Save the quantized model
+mlpack::data::Save("quantized_model.bin", "model", model);
+
+// Use the quantized model for inference
+arma::mat testData; // Load your test data
+arma::mat predictions;
+model.Predict(testData, predictions);
+```
+<p style="text-align: center; font-size: 85%"><a href="#simple-examples">More examples...</a></p>
+
+#### Quick links:
+
+ * [Constructors](#constructors): create `Quantization` objects.
+ * [`Apply()`](#applying-quantization): apply quantization to model or weights.
+ * [Other functionality](#other-functionality) for customizing quantization parameters.
+ * [Examples](#simple-examples) of simple usage and links to detailed example projects.
+ * [Template parameters](#advanced-functionality-template-parameters) for custom behavior.
+ * [Advanced template examples](#advanced-functionality-examples) of use with custom template parameters.
+
+#### See also:
+
+ * [`FFN`](ffn.md): feed-forward neural networks
+ * [`CNN`](cnn.md): convolutional neural networks
+ * [mlpack transformations](../../index.md#transformations)
+ * [Quantization on Wikipedia](https://en.wikipedia.org/wiki/Quantization_(signal_processing))
+ * [Quantization for Neural Networks](https://arxiv.org/abs/1712.05877) (comprehensive overview paper, pdf)
+
+### Constructors
+
+Construct a `Quantization` object using one of the constructors below. Defaults and types are detailed in the [Constructor Parameters](#constructor-parameters) section below.
+
+#### Forms:
+
+ * `q = Quantization()`
+   - Initialize quantizer with default settings (8-bit linear quantization).
+
+---
+
+ * `q = Quantization(bits)`
+   - Initialize quantizer with specified bit depth.
+
+---
+
+#### Constructor Parameters:
+
+| **name** | **type** | **description** | **default** |
+|----------|----------|-----------------|-------------|
+| `bits` | `size_t` | Number of bits to use for quantization. | `8` |
+
+### Applying Quantization
+
+Once a `Quantization` object is created, the `Apply()` member function can be used to quantize models or weight matrices.
```

Comment:
Thanks, the documentation here is nice. But I don't think we intend for a user to directly use a `Quantization` object.  Instead they can call `FFN::Quantize()`.  So really the documentation should (in my opinion) not treat `Quantization` as its own object, but instead be a small-ish subsection (including the details about custom quantization strategies) in the FFN documentation.

But, I realize, the FFN documentation doesn't exist yet.  So, I think one option here is to leave this mostly as-is (maybe simplify a bit) and simply don't link any other documentation to this page for now. Then, when I add the FFN documentation, I will work this in.

Let me know what you think.